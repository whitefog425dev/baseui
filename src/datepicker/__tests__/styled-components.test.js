/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {
  StyledRoot,
  StyledCalendarContainer,
  StyledHeader,
  StyledMonthHeader,
  StyledMonth,
  StyledWeek,
  StyledDay,
} from '../index.js';

const styledComponents = [
  [StyledRoot, 'StyledRoot'],
  [StyledCalendarContainer, 'StyledCalendarContainer'],
  [StyledHeader, 'StyledHeader'],
  [StyledMonthHeader, 'StyledMonthHeader'],
  [StyledMonth, 'StyledMonth'],
  [StyledWeek, 'StyledWeek'],
  [StyledDay, 'StyledDay'],
];

const props = [
  ['$disabled'],
  ['$isHighlighted'],
  ['$isHovered'],
  ['$outsideMonth'],
  ['$selected'],
];

describe('Component styled components', () => {
  test.each(styledComponents)('Basic render', (Component, name) => {
    const component = shallow(<Component />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has correct styles`,
    );
  });

  test.each(props)('Prop based styles', prop => {
    const component = shallow(<StyledDay {...{[prop]: true}} />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `StyledDay has correct styles when ${prop} is set to true`,
    );
  });
});
