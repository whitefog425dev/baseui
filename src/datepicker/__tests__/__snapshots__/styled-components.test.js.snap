// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component styled components Basic render: StyledCalendarContainer has correct styles 1`] = `
Object {
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale400",
}
`;

exports[`Component styled components Basic render: StyledDay has correct styles 1`] = `
Object {
  "backgroundColor": "transparent",
  "borderRadius": "$theme.borders.radius200",
  "boxSizing": "border-box",
  "color": "inherit",
  "cursor": "pointer",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "lineHeight": "$theme.sizing.scale800",
  "marginBottom": "0",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "0",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale200",
  "paddingRight": "$theme.sizing.scale200",
  "paddingTop": "$theme.sizing.scale300",
  "textAlign": "center",
  "width": "$theme.sizing.scale1000",
}
`;

exports[`Component styled components Basic render: StyledHeader has correct styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "$theme.colors.primary",
  "borderRadius": "$theme.borders.radius200 $theme.borders.radius200 0 0",
  "color": "$theme.colors.white",
  "display": "flex",
  "justifyContent": "space-between",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale500",
}
`;

exports[`Component styled components Basic render: StyledMonth has correct styles 1`] = `
Object {
  "display": "inline-block",
}
`;

exports[`Component styled components Basic render: StyledMonthHeader has correct styles 1`] = `
Object {
  "whiteSpace": "no-wrap",
}
`;

exports[`Component styled components Basic render: StyledRoot has correct styles 1`] = `
Object {
  "backgroundColor": "$theme.colors.white",
  "borderRadius": "$theme.borders.radius200",
  "color": "$theme.colors.black",
  "display": "inline-block",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "textAlign": "center",
}
`;

exports[`Component styled components Basic render: StyledWeek has correct styles 1`] = `
Object {
  "display": "flex",
  "whiteSpace": "no-wrap",
}
`;

exports[`Component styled components Prop based styles: StyledDay has correct styles when $disabled is set to true 1`] = `
Object {
  "backgroundColor": "transparent",
  "borderRadius": "$theme.borders.radius200",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono600",
  "cursor": "default",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "lineHeight": "$theme.sizing.scale800",
  "marginBottom": "0",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "0",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale200",
  "paddingRight": "$theme.sizing.scale200",
  "paddingTop": "$theme.sizing.scale300",
  "textAlign": "center",
  "width": "$theme.sizing.scale1000",
}
`;

exports[`Component styled components Prop based styles: StyledDay has correct styles when $isHighlighted is set to true 1`] = `
Object {
  "backgroundColor": "$theme.colors.primary100",
  "borderRadius": "$theme.borders.radius200",
  "boxSizing": "border-box",
  "color": "inherit",
  "cursor": "pointer",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "lineHeight": "$theme.sizing.scale800",
  "marginBottom": "0",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "0",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale200",
  "paddingRight": "$theme.sizing.scale200",
  "paddingTop": "$theme.sizing.scale300",
  "textAlign": "center",
  "width": "$theme.sizing.scale1000",
}
`;

exports[`Component styled components Prop based styles: StyledDay has correct styles when $isHovered is set to true 1`] = `
Object {
  "backgroundColor": "$theme.colors.primary100",
  "borderRadius": "$theme.borders.radius200",
  "boxSizing": "border-box",
  "color": "inherit",
  "cursor": "pointer",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "lineHeight": "$theme.sizing.scale800",
  "marginBottom": "0",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "0",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale200",
  "paddingRight": "$theme.sizing.scale200",
  "paddingTop": "$theme.sizing.scale300",
  "textAlign": "center",
  "width": "$theme.sizing.scale1000",
}
`;

exports[`Component styled components Prop based styles: StyledDay has correct styles when $outsideMonth is set to true 1`] = `
Object {
  "backgroundColor": "transparent",
  "borderRadius": "$theme.borders.radius200",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono600",
  "cursor": "pointer",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "lineHeight": "$theme.sizing.scale800",
  "marginBottom": "0",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "0",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale200",
  "paddingRight": "$theme.sizing.scale200",
  "paddingTop": "$theme.sizing.scale300",
  "textAlign": "center",
  "width": "$theme.sizing.scale1000",
}
`;

exports[`Component styled components Prop based styles: StyledDay has correct styles when $selected is set to true 1`] = `
Object {
  "backgroundColor": "$theme.colors.primary",
  "borderRadius": "$theme.borders.radius200",
  "boxSizing": "border-box",
  "color": "$theme.colors.white",
  "cursor": "pointer",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "lineHeight": "$theme.sizing.scale800",
  "marginBottom": "0",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "0",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale200",
  "paddingRight": "$theme.sizing.scale200",
  "paddingTop": "$theme.sizing.scale300",
  "textAlign": "center",
  "width": "$theme.sizing.scale1000",
}
`;
